### ISSUE TITLE
Parallel test failure from checkpoint file collisions

### ISSUE THREAD

**Issue by wthrowe** [sxs-collaboration/spectre#4096]
Various `Unit.Parallel` tests can fail when run in parallel with the error
```
Node: 0 Proc: 0
Line: 857 of /home/wthrowe/spectre/src/Parallel/Main.hpp
Function: void Parallel::Main<TestMetavariables>::check_future_checkpoint_dirs_available() const [Metavariables = TestMetavariables]
Can't start run: found checkpoints that may be overwritten!
Dirs from SpectreCheckpoint000000 onward must not exist.

############ ERROR ############
```

This was introduced in 543754d889082a0104ce02febffdca2e7ce046c2 because the check for checkpoint directories is dependent on the existence of the `WriteCheckpoint` phase, which after that commit always exists but was previously absent from most testing executables.

**Comment by kidder** [comment]
I have a temporary fix in #4098 

This is really two issues, one of which existed before the changes to how phases are handled:

1.  The old issue is that tests run in parallel can be run in the same directory and clash.  This is why we have different names for h5 files.  If there had been two tests that wrote checkpoint files, they would have clashed.  My temporary fix only addresses this for checkpoints generated through the `add_algorithm_test_with_restart_from_checkpoint` cmake function
2. The new issue is that all executables now fail if checkpoint files exist instead of just those with an explicit WriteCheckpoint phase.   I will need to investigate whether Main has enough information to determine whether PhaseControl can or cannot choose the WriteCheckpoint phase...
